modules/core/plan/src/Access/PlanRecordAccess.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Drupal\plan\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityAccessControlHandler;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Defines plan_record access logic.
+ */
+class PlanRecordAccess extends EntityAccessControlHandler {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function checkAccess(EntityInterface $entity, $operation, AccountInterface $account) {
+
+    // If a plan is referenced, access is based on access to the plan.
+    /** @var \Drupal\plan\Entity\PlanInterface[] $plans */
+    $plans = $entity->get('plan')->referencedEntities();
+    if (!empty($plans[0])) {
+      return AccessResult::allowedIf($plans[0]->access($operation, $account));
+    }
```

Comment:
I was going to mention this while reviewing the Crop plan code - I think there is an easier way to reference the first referenced entity on entity reference fields using the `->first()` and `->entity()` methods. I think this can refined a bit further using the "nullsafe operator" `?->` I've just learned about, too.

```suggestion
    if ($plan = $entity->get('plan')->first()?->entity()) {
      return AccessResult::allowedIf($plan->access($operation, $account));
    }
```


And actually, thinking more generally and re: the crop plan.... it would be nice to include this logic as a helper method on the `PlanRecord` entity class itself. Then you could just do `if ($plan = $entity->getPlan())`

Reply:
I've implemented this in my `3.x-plan-record` branch: https://github.com/paul121/farmOS/commit/0a0c28a1ce9616cd752d94036db9f43feb4714b2

Reply:
I love this idea - thanks @paul121! I merged in your commit.